src/endpoint/s3/s3_utils.js

```diff
@@ -785,6 +785,30 @@ function key_marker_to_cont_tok(key_marker, objects_arr, is_truncated) {
     return Buffer.from(j).toString('base64');
 }
 
+/**
+ * Returns true if the byte length of the key
+ * is within the range [0, max_length]
+ * @param {string} key 
+ * @param {number} max_length 
+ * @returns 
+ */
+function verify_string_byte_length(key, max_length) {
+    // If max_length is infintiy then no need to do
+    // any further checks
+    if (!Number.isFinite(max_length)) {
```

Comment:
This check is covered by the next check of len*4<=max, so I would remove this one just to reduce redundant logic (which is what the Infinity value was meant for)